transformations/transformations/scc_cuf.py

```diff
@@ -0,0 +1,867 @@
+    routine.body.insert(insert_index, ir.Comment(''))
+
+    # TODO: this just assumes that host-device-synchronisation is only needed at the beginning and end
+    # Copy device to host
+    insert_index = None
+    for call in FindNodes(ir.CallStatement).visit(routine.body):
+        if "THREAD_END" in str(call.name):
+            insert_index = routine.body.body.index(call) + 1
+
+    if insert_index is None:
+        routine.body.append(ir.Comment(''))
+        routine.body.append(ir.Comment('! Copy device to host'))
+    for v in reversed(relevant_arrays):
+        if v.type.intent != "in":
+            lhs = v.clone(dimensions=None)
+            vtype = v.type.clone(device=True, allocatable=True, intent=None, shape=None)
+            rhs = v.clone(name=f"{v.name}_d", type=vtype, dimensions=None)
+            if insert_index is None:
+                routine.body.append(ir.Assignment(lhs=lhs, rhs=rhs))
+            else:
+                routine.body.insert(insert_index, ir.Assignment(lhs=lhs, rhs=rhs))
+    if insert_index is not None:
+        routine.body.insert(insert_index, ir.Comment('! Copy device to host'))
+
+    # De-allocation
+    routine.body.append(ir.Comment(''))
+    routine.body.append(ir.Comment('! De-allocation'))
+    for array in relevant_arrays:
+        routine.body.append(ir.Deallocation((array.clone(name=f"{array.name}_d", dimensions=None),)))
+
+    call_map = {}
+    for call in calls:
+        arguments = []
+        for arg in call.arguments:
+            if arg in relevant_arrays:  # if isinstance(arg, sym.Array):
+                vtype = arg.type.clone(device=True, allocatable=True, shape=None, intent=None)
+                arguments.append(arg.clone(name=f"{arg.name}_d", type=vtype, dimensions=None))
+            else:
+                arguments.append(arg)
+        call_map[call] = call.clone(arguments=arguments)
+    routine.body = Transformer(call_map).visit(routine.body)
+
+
+def driver_launch_configuration(routine, block_dim, disable):
+    """
+    Launch configuration for kernel calls within the driver with the
+    CUDA Fortran (CUF) specific chevron syntax `<<<griddim, blockdim>>>`.
+
+    Parameters
+    ----------
+    routine: :any:`Subroutine`
+        The subroutine to specify the launch configurations for kernel calls.
+    block_dim: :any:`Dimension`
+        The dimension object specifying the block loop dimension
+    disable: tuple
+        Tuple of routines not to be processed
+    """
+
+    d_type = SymbolAttributes(types.DerivedType("DIM3"))
+    routine.spec.append(ir.VariableDeclaration(symbols=(sym.Variable(name="GRIDDIM", type=d_type),
+                                                        sym.Variable(name="BLOCKDIM", type=d_type))))
+
+    loop_map = {}
+    call_map = {}
+    for loop in FindNodes(ir.Loop).visit(routine.body):
+        if loop.variable == block_dim.index or loop.variable in block_dim._aliases:
```

Comment:
I'm confused by this: `size_expressions` are declaration dimensions, while `loop.variable` should be a `dim.index`. We don't have indexing for dimensions yet, so this might be a necessary workaround for now.